lib/plug/upload.ex

```diff
@@ -59,25 +59,66 @@ defmodule Plug.Upload do
     end
   end
 
+  @doc """
+  Assign ownership of the given upload file to another process.
+
+  Useful if you want to do some work on an uploaded file in another process
+  since it means that the file will survive the end of the request.
+  """
+  @spec give_away(t | binary, pid, pid) :: :ok | {:error, binary}
+  def give_away(path, to_pid, from_pid \\ self())
+
+  def give_away(%__MODULE__{path: path}, to_pid, from_pid) do
+    give_away(path, to_pid, from_pid)
+  end
+
+  def give_away(path, to_pid, from_pid)
+      when is_binary(path) and is_pid(to_pid) and is_pid(from_pid) do
+    case :ets.lookup(@table, from_pid) do
+      [{^from_pid, _tmp, paths}] ->
+        updated_paths = Enum.reject(paths, &(&1 == path))
+
+        case :ets.lookup(@table, to_pid) do
+          [{^to_pid, _tmp, paths}] ->
+            true = :ets.update_element(@table, to_pid, {3, [path | paths]})
```

Comment:
These operations need to happen on the server, otherwise you will have race conditions. :)

Reply:
Doh. Ok. Will rearrange.. Thanks

Reply:
For example, if two transfers to the same process happen at the same time.